starter-code/src/functions-and-arrays.js

```diff
@@ -99,3 +163,47 @@ const matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+function greatestProduct (array) {
```

Comment:
Very diligent of you to search for diagonal combinations as well. Awesome job :)